state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -230,6 +246,81 @@ pub mod pallet {
 		UtxoConsolidationParametersUpdated { params: utxo_selection::ConsolidationParameters },
 		/// Arbitrum Initialized: contract addresses have been set, first key activated
 		ArbitrumInitialized,
+		/// Some Utxos has been sent from the previous vault into the current vault.
+		UtxoTransferred { broadcast_id: BroadcastId, total_amount: BtcAmount },
+		/// Some Utxo from before the previous epochs are discarded from storage.
+		StaleUtxoDiscarded { utxos: Vec<Utxo> },
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// Sweep the current utxo storage and:
+		/// 1. Transfer a limited number of utxos from the previous vault into the current vault.
+		/// 2. Any utxos from earlier than the previous vault are discarded.
+		fn on_finalize(_block_number: BlockNumberFor<T>) {
```

Comment:
Is there any reason we'd leave the current consolidation logic in the ingress egress pallet? There are two different scenarios:
1. Consolidate because we have a lot of UTXOs (sending from vault A to Vault A)
2. Rotate move over all the utxos to the new vault (sending from Vault A to Vault B).

However, they are very closely related, and keeping these two "automatic" utxo sending code paths in the same place would make this a bit easier to follow imo.